crates/rspack_plugin_javascript/src/dependency/context/mod.rs

```diff
@@ -72,3 +75,46 @@ fn context_dependency_template_as_require_call(
   }
   source.replace(callee_start, callee_end, &expr, None);
 }
+
+fn context_dependency_template_as_id(
+  dep: &dyn ContextDependency,
+  source: &mut TemplateReplaceSource,
+  code_generatable_context: &mut TemplateContext,
+  range: &RealDependencyLocation,
+) {
+  let TemplateContext {
+    compilation,
+    runtime_requirements,
+    ..
+  } = code_generatable_context;
+  let id = dep.id();
+
+  let expr = module_raw(
+    compilation,
+    runtime_requirements,
+    id,
+    dep.request(),
+    dep.options().mode == ContextMode::Weak,
+  );
+
+  if compilation
+    .get_module_graph()
+    .module_graph_module_by_dependency_id(id)
+    .is_none()
+  {
+    source.replace(range.start, range.end, &expr, None);
+    return;
+  }
+
+  for (content, start, end) in &dep.options().replaces {
+    source.replace(*start, *end - 1, content, None);
```

Comment:
This should be also not needed, the range returned by create_context_dependency is already -1

Reply:
https://github.com/web-infra-dev/rspack/blob/146e0be6279376249e3502a88214f2d80d87a213/crates/rspack_plugin_javascript/src/dependency/context/mod.rs#L74

It doesn't seem so.

Reply:
I see..